integrations/lambda/notebooks/demo.ipynb

```diff
@@ -0,0 +1,442 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6efcc07e",
+   "metadata": {},
+   "source": [
+    "# Timestream Lambda Function Sample Application"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dfb5f3c3",
+   "metadata": {},
+   "source": [
+    "This notebook demonstrates generating data, according to a schema defined by the user; deploying an AWS Lambda function to process it; and visualizing the data using Grafana."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c1b993b8",
+   "metadata": {},
+   "source": [
+    "## Imports\n",
+    "\n",
+    "Imports necessary for all steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "70bf2bbf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "from datetime import datetime, timedelta, timezone\n",
+    "import json\n",
+    "import math\n",
+    "import sys\n",
+    "\n",
+    "sys.path.append('../src/')\n",
+    "from lambda_sample import data_generator, lambda_helper, grafana_helper"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "85d6d108",
+   "metadata": {},
+   "source": [
+    "# AWS Settings\n",
+    "\n",
+    "These constants and variables are used throughout this notebook to configure writes to Timestream for LiveAnalytics and the generation of Amazon Managed Grafana dashboards.\n",
+    "\n",
+    "Adjust these settings as needed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "2da0e9c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DATABASE_NAME = \"sample_app_database\"\n",
+    "TABLE_NAME = \"sample_app_table\"\n",
+    "\n",
+    "# To be used later, by the Lambda function, to create the Timestream for LiveAnalytics table.\n",
+    "# If you created your table manually, update with the actual values you configured for your table.\n",
+    "# Default values when creating a new table in the AWS console.\n",
+    "MEM_STORE_RETENTION_PERIOD_IN_HOURS = 12\n",
+    "MAG_STORE_RETENTION_PERIOD_IN_DAYS = 3653 # 10 years\n",
+    "\n",
+    "# The number of records to ingest to Timestream for LiveAnalytics at a time.\n",
+    "# Timestream for LiveAnalytics accepts a maximum of 100 records at a time.\n",
+    "BATCH_SIZE = 100\n",
+    "# BATCH_SIZE = 1\n",
+    "\n",
+    "# The precision of the timestamp for each generated record. Valid options are \"MILLISECONDS\", \"SECONDS\", and \"MICROSECONDS\".\n",
```

Comment:
Timestream records support nanoseconds as well. Should we support this precision as well?

Reply:
This is due to a limitation in `datetime`, it doesn't support nanosecond precision without multiplying microseconds by a factor of 10, which is useless in my opinion.